### DISCUSSION TITLE
How do I run CLBlas, the android guide seems a bit unclear for the last steps.

### DISCUSSION THREAD

**Discussion by Avenir2233** [ggml-org/llama.cpp#1951] [Q&A]
I have been trying to follow the guide for running the the code with android, and have had an issue or two. https://github.com/ggerganov/llama.cpp/commit/bac19927c302737465a1deb14ac0943a221863e8

For lines 666-669, what exactly should I be doing with? Should I use it in my main CLBlast folder, my com.termux/files/home folder, or somewhere else? Also I don't know if its related but when I tried to run it somewhere I got for the ./main (...) code it returns a bash: syntax error near unexpected token '...' . Is there somethin I'm missing? Because up to that part everything seemed to install correctly.

**Comment by ghost** [comment]
> I have been trying to follow the guide for running the the code with android, and have had an issue or two. [bac1992](https://github.com/ggerganov/llama.cpp/commit/bac19927c302737465a1deb14ac0943a221863e8)
> 
> For lines 666-669, what exactly should I be doing with? Should I use it in my main CLBlast folder, my com.termux/files/home folder, or somewhere else? Also I don't know if its related but when I tried to run it somewhere I got for the ./main (...) code it returns a bash: syntax error near unexpected token '...' . Is there somethin I'm missing? Because up to that part everything seemed to install correctly.

Hi, I agree that section of the Android guide needs help. 

I assume that you've successfully installed CLBlast and can run clinfo? If clinfo receives a No Platform message then OpenCL and CLBlast is not functioning correctly on your system.

Consider running clinfo (you can run it anywhere), but try this:
```
$HOME

LD_LIBRARY_PATH=/vendor/lib64:$PREFIX/lib clinfo
```

If you receive a platform, and it displays your OpenCL details, then it's working.  Run ./main from llama.cpp/build/bin, with something like this:
`
LD_LIBRARY_PATH=/vendor/lib64:$PREFIX/lib ./main (...)
`

Let me know how it goes.
